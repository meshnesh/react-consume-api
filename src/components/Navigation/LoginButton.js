import React from 'react';
import { Link } from 'react-router-dom';
import ModalBUtton from '../Modal';



export function LoginButton() {
	return (
		<Link to="/register">
			Register
		</Link>
	);
}

export function LogoutButton(props) {
	return (
		<a onClick={props.onClick}>
			Log out
		</a>
	);
}
