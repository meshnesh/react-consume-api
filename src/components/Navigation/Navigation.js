import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';



import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import Menu from 'material-ui/Menu';
import MenuItem from 'material-ui/MenuItem';
import ViewModule from 'material-ui/svg-icons/action/view-module';
import {white} from 'material-ui/styles/colors';

import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import { userLogoutActions, userLoginActions } from '../../actions/user.actions';

class Nav extends Component {

	constructor(props) {
		super(props);
		this.handleLogoutClick = this.handleLogoutClick.bind(this);
		this.state = {
			isLoggedIn: this.props.userInfo.isLoggedIn
		};
	}


	handleLogoutClick() {
		this.setState({isLoggedIn: false});
		this.props.userLogoutActions();
	}


	render() {

		const isLoggedIn = this.state.isLoggedIn;

		const button = isLoggedIn ? (
			<div>
				<Link to="/newProfile">
					<MenuItem key={1} primaryText="Profile"/>
				</Link>

				<MenuItem primaryText="Sign out" onClick={this.handleLogoutClick} />
			</div>
		) : (
			<Link to="/Login">
				<MenuItem key={1} primaryText="Login"/>
			</Link>
		);


		return (
			<nav>
				<div className="nav-wrapper">
					<Link className="brand-logo" to="/">
                        Bright Events
					</Link>
                    
					<ul id="nav-mobile hide-on-small-only" className="right hide-on-med-and-down">

						<IconMenu color={white}
							iconButtonElement={
								<IconButton><MoreVertIcon color={white}/></IconButton>
							}
							targetOrigin={{horizontal: 'right', vertical: 'top'}}
							anchorOrigin={{horizontal: 'right', vertical: 'top'}}
						>
							{ button }
						</IconMenu>
					</ul>
				</div>
			</nav>
		);
	} 
}

const mapStateToProps = state => {
	return{
		userInfo:state.authLogin
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		userLogoutActions: () =>dispatch(userLogoutActions())
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
