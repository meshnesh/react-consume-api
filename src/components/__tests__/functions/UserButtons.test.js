import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { LogoutButton } from '../../../components/Navigation/LoginButton';

describe('Button', () => {
	const mockFn = jest.fn();
	it('should be defined', () => {
		expect(LogoutButton).toBeDefined();
	});
    
	it('should render correctly', () => {
		const tree = shallow(
			<LogoutButton name='button test' handleClick={mockFn} />
		);
		expect(tree).toMatchSnapshot();
	});

});
