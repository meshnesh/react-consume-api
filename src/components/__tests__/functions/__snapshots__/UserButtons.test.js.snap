// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogoutButton
    handleClick={[MockFunction]}
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Log out",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "Log out",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Log out",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "Log out",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
